### PR TITLE
Add spider for century_21

### PR DESCRIPTION
**Pull request by abbensid2**

### CODE REVIEW

**Comment by Cj-Malone** [comment]
Could this spider be turned (almost) global. If you start with the [sitemap](https://www.century21global.com/sitemap.xml) and follow all the `/real-estate-offices/` links?

**Comment by Cj-Malone** [comment]
It looks like some of them also have coords available with `.//@data-lat` and `.//@data-lng`

File: locations/spiders/century_21.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                ref=id,
+                addr_full=address,
+                country=country,
+                lat=lat,
+                lon=lon,
+                website=f"https://www.{self.allowed_domains[0]}{url}",
+            )
+
+        # Pagination next page
+        if response.xpath('//a[contains(@aria-label, "Siguiente")]/@href').get():
```
Comment (Cj-Malone) [inline locations/spiders/century_21.py:37]: This is now using English URLs, so we need to update the next label. Also I think you meant to assign a variable here.
```suggestion
        if next := response.xpath('//a[contains(@aria-label, "Next")]/@href').get():
```

File: locations/spiders/century_21.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                addr_full=address,
+                country=country,
+                lat=lat,
+                lon=lon,
+                website=f"https://www.{self.allowed_domains[0]}{url}",
+            )
+
+        # Pagination next page
+        if next := response.xpath('//a[contains(@aria-label, "Next")]/@href').get():
+            yield scrapy.Request(url=f"https://www.{self.allowed_domains[0]}{next}", callback=self.parse)
```
Comment (iandees) [inline locations/spiders/century_21.py:38]:
```suggestion
            yield scrapy.Request(url=request.urljoin(next), callback=self.parse)
```

**Review by Cj-Malone — APPROVED** [review]
Good enough for me, thank you.

**Pull request merged**